x-pack/legacy/plugins/code/public/utils/saga_utils.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { take, spawn } from 'redux-saga/effects';
+import { Action } from 'redux-actions';
+
+function* cancelRequest(action: Action<any>, abortController: AbortController) {
+  yield take(action.type);
+  abortController.abort();
+}
+
+export const cancellableSaga = (saga: any) =>
```

Comment:
nit: this sounds more like a **singleton** saga, instead of a **cancellable** saga. After all, every saga is always **cancellable** by itself. 😀